pkg/buildpipeline/pipeline_test.go

```diff
@@ -1322,3 +1323,93 @@ func TestRfc1035LabelMangling(t *testing.T) {
 		})
 	}
 }
+
+func testFindDuplicates(t *testing.T) {
```

Comment:
nit: I'd just add these to the general `TestParseJenkinsfileYaml` tests so we get full validation on them as well.

Reply:
good point, to test the expected error message

Reply:
Yeah, also it just gives us more cases to make sure we aren't accidentally breaking things when we change other stuff in validation in the future.